src/main/webapp/app/exam/manage/exams/exam-exercise-import/exam-exercise-import.component.ts

```diff
@@ -78,14 +85,21 @@ export class ExamExerciseImportComponent implements OnInit {
         this.exam.exerciseGroups?.forEach((exerciseGroup) => {
             this.selectedExercises.set(exerciseGroup, new Set<Exercise>(exerciseGroup.exercises));
         });
+        const duplicated = new Set<string>();
         // Initialize containsProgrammingExercises
         this.exam.exerciseGroups!.forEach((exerciseGroup) => {
             const hasProgrammingExercises = !!exerciseGroup.exercises?.some((value) => value.type === ExerciseType.PROGRAMMING);
             this.containsProgrammingExercises.set(exerciseGroup, hasProgrammingExercises);
             // In case of a rejected import, we can delete programming exercises with a title from the Map / blocklist, as those were not rejected by the server.
             exerciseGroup.exercises?.forEach((exercise) => {
                 if (exercise.type === ExerciseType.PROGRAMMING && exercise.title) {
-                    this.titleAndShortNameOfProgrammingExercises.delete(exercise.id!);
+                    if (!duplicated.has(exercise.title)) {
+                        duplicated.add(exercise.title);
+                        this.titleAndShortNameOfProgrammingExercises.delete(exercise.id!);
+                    } else {
+                        this.titleAndShortNameOfProgrammingExercises.set(exercise.id!, [exercise.title!, ``]);
```

Comment:
Why do you create this array here? Can you not just call `set(id, title);`?

Reply:
Yes exactly, short name is used for when a programming exercise with the same project name already exists on Bamboo. In that case the server returnes an empty title and short name of the corresponding exercise and the previous title and short name are used as placeholder. A normal string is used now :+1: